rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/60_knn_search_filter_alias.yml

```diff
@@ -0,0 +1,107 @@
+setup:
+  - skip:
+      version: ' - 8.4.99'
+      reason: 'filtered alias for kNN search added in 8.5'
+      
+  - do:
+      indices.create:
+        index: test
+        body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
+          mappings:
+            dynamic: false
+            properties:
+              test_vector:
+                type: dense_vector
+                dims: 4
+                index : true
+                similarity : l2_norm
+              name:
+                type: keyword
+                store: true
+
+  - do:
+      index:
+        index: test
+        id: "1"
+        body:
+          name: v1
+          test_vector: [230.0, 300.33, -34.8988, 15.555]
+
+  - do:
+      index:
+        index: test
+        id: "2"
+        body:
+          name: v1
+          test_vector: [0.5, 0.5, 0.5, -1]
+
+  - do:
+      index:
+        index: test
+        id: "3"
+        body:
+          name: v2
+          test_vector: [0.5, 0.5, 0.5, 0.5]
+
+  - do:
+      index:
+        index: test
+        id: "4"
+        body:
+          name: v2
+          test_vector: [0.5, 0.5, 0.5, 0]
+
+  - do:
+      indices.put_alias:
+        index: test
+        name: test-alias
+        body:
+          filter:
+            term:
+              name: v1
+
+  - do:
+     indices.refresh: {}
+
+---
+"kNN filter alias":
+
+  - do:
+      search:
+        index: test
+        body:
+          fields: [ name ]
+          knn:
+            field: test_vector
+            query_vector: [ 230.0, 300.33, -34.8988, 15.555 ]
+            k: 2
+            num_candidates: 100
+            filter:
+              term:
+                name: v1
+
+  - match: { hits.hits.0._id: "1" }
+  - match: { hits.hits.0.fields.name.0: v1 }
+
+  - match: { hits.hits.1._id: "2" }
+  - match: { hits.hits.1.fields.name.0: v1 }
+
```

Comment:
may be worth providing a small comment explaining what's happening,  or may be making vector values more obvious.
Something like: if there was no proper pre-filter implemented for vector search, the closest docs returned from vector search are with ids of  `1` and `3`, and the post-filter on them will return only a doc with id:`1`. With proper pre-filter, we ensure that filter is applied during knn search thus returning 1 result.

Reply:
@mayya-sharipova I added the additional test you suggested where the first search w/o the filtered alias shows the results where vectors 1 and 3 are nearest where 1 has the term v1 and 3 has the term v2. I then added the filtered alias and the results change to vectors 1 and 2 where those are the two nearest with the term v1. Is this the correct behavior since k = 2 or should the filter only return a single result since 1 and 2 are not technically the two nearest vectors without it?

Reply:
@jdconrad Thanks, that's correct behaviour; we need to return the nearest vectors that satisfy the filter (so it should  ids: 1 and 2 )